presto-hive/src/main/java/com/facebook/presto/hive/parquet/dictionary/ParquetDictionaryReader.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.hive.parquet.dictionary;
+
+import parquet.bytes.BytesUtils;
+import parquet.column.values.ValuesReader;
+import parquet.column.values.rle.RunLengthBitPackingHybridDecoder;
+import parquet.io.ParquetDecodingException;
+import parquet.io.api.Binary;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+public class ParquetDictionaryReader
+    extends ValuesReader
+{
+    private ByteArrayInputStream in;
+    private ParquetDictionary dictionary;
+    private RunLengthBitPackingHybridDecoder decoder;
+
+    public ParquetDictionaryReader(ParquetDictionary dictionary)
+    {
+        this.dictionary = dictionary;
+    }
+
+    @Override
+    public void initFromPage(int valueCount, byte[] page, int offset)
+        throws IOException
+    {
+        checkArgument(page.length > offset, "Attempt to read from empty page");
```

Comment:
Is `page.length > offset` really an empty page condition as the error message says? Seems more like offset is not in the page.